vcr/matchers.py

```diff
@@ -30,12 +32,29 @@ def query(r1, r2):
     return r1.query == r2.query
 
 
-def body(r1, r2):
+def raw_body(r1, r2):
     if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
         return r1.body.read() == r2.body.read()
     return r1.body == r2.body
 
 
+def body(r1, r2):
+    if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
+        r1_body = r1.body.read()
+        r2_body  = r2.body.read()
+    else:
+        r1_body = r1.body
+        r2_body  = r2.body
+    if r1.headers.get('Content-Type') == r2.headers.get('Content-Type') == 'application/x-www-form-urlencoded':
```

Comment:
> But yeah, the header comparison should probably be case-insensitive for our "smart body matcher". Just don't store them that way.

Oh yeah, agreed. What I was saying is that we can't check for the 'Content-Type' key or assume that 'application/x-www-form-urlencoded' will always be totally lower cased, and so that here we should take request.headers and insert it into a case insensitve dict in order to check for the header.